src/hotspot/share/gc/shared/gcLocker.hpp

```diff
@@ -27,126 +27,30 @@
 
 #include "gc/shared/gcCause.hpp"
 #include "memory/allStatic.hpp"
-#include "utilities/globalDefinitions.hpp"
-#include "utilities/macros.hpp"
-
-class JavaThread;
-
-// The direct lock/unlock calls do not force a collection if an unlock
-// decrements the count to zero. Avoid calling these if at all possible.
+#include "runtime/mutex.hpp"
 
```

Comment:
Documentation how GCLocker works/is supposed to work is missing here. It's not exactly trivial.